LethalAPI.Core/Events/Features/Patcher.cs

```diff
@@ -0,0 +1,160 @@
+// -----------------------------------------------------------------------
+// <copyright file="Patcher.cs" company="LethalAPI Modding Community">
+// Copyright (c) LethalAPI Modding Community. All rights reserved.
+// Licensed under the GPL-3.0 license.
+// </copyright>
+// Taken from EXILED (https://github.com/Exiled-Team/EXILED)
+// Licensed under the CC BY SA 3 license. View it here:
+// https://github.com/Exiled-Team/EXILED/blob/master/LICENSE.md
+// Changes: Namespace adjustments.
+// -----------------------------------------------------------------------
+
+namespace LethalAPI.Core.Events.Features;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+using Attributes;
+using Interfaces;
+
+/// <summary>
+/// A tool for patching.
+/// </summary>
+public class Patcher
+{
+    /// <summary>
+    /// The below variable is used to increment the name of the harmony instance, otherwise harmony will not work upon a plugin reload.
+    /// </summary>
+    // ReSharper disable once InconsistentNaming
+    private static int patchesCounter;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Patcher"/> class.
+    /// </summary>
+    internal Patcher()
+    {
+        Harmony = new($"exiled.events.{++patchesCounter}");
+    }
+
+    /// <summary>
+    /// Gets a <see cref="HashSet{T}"/> that contains all patch types that haven't been patched.
+    /// </summary>
+    // ReSharper disable once MemberCanBePrivate.Global
+    public static HashSet<Type> UnpatchedTypes { get; private set; } = Events.UseDynamicPatching ? GetNonEventPatchTypes() : GetAllPatchTypes();
+
+    /// <summary>
+    /// Gets a set of types and methods for which EXILED patches should not be run.
```

Comment:
More Exiled mentions here

Reply:
I'll do a mass-Exiled Purge here in a sec.